File: app/src/main/java/mchehab/com/realmdemo/MainActivity.java
package mchehab.com.realmdemo;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;



import io.realm.Realm;

public class MainActivity extends AppCompatActivity {

    private EditText editTextFirstName;
    private EditText editTextLastName;
    private EditText editTextAge;
    private Button buttonSave;
    private Button buttonRestart;

    private Person person = new Person();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editTextFirstName = findViewById(R.id.editTextFirstName);
        editTextLastName = findViewById(R.id.editTextLastName);
        editTextAge = findViewById(R.id.editTextAge);
        buttonSave = findViewById(R.id.button);
        buttonRestart = findViewById(R.id.buttonRestart);

        if(isPersonSaved()){
            person = getPerson();
            buttonSave.setText("Update");
        }

        editTextFirstName.setText(person.getFirstName());
        editTextLastName.setText(person.getLastName());
        if(person.getAge() > 0){
            editTextAge.setText(person.getAge() + "");
        }

        buttonSave.setOnClickListener(e -> {
            Realm realm = Realm.getDefaultInstance();
            realm.beginTransaction();

            person.setFirstName(editTextFirstName.getText().toString());
            person.setLastName(editTextLastName.getText().toString());
            person.setAge(Integer.parseInt(editTextAge.getText().toString()));

            if(buttonSave.getText().toString().equalsIgnoreCase("save")){
                realm.insert(person);
            }
            realm.commitTransaction();
            realm.close();
        });

        buttonRestart.setOnClickListener(e -> {
            Intent intent = getIntent();
            finish();
            startActivity(intent);
        });
    }

    private boolean isPersonSaved(){
        Realm.init(this);
        return Realm.getDefaultInstance().where(Person.class).findFirst() != null;
    }

    private Person getPerson(){
        Realm.init(this);
        Realm realm = Realm.getDefaultInstance();
        return realm.where(Person.class).findFirst();
    }
}

File: kotlinapp/src/main/java/mchehab/com/kotlinapp/MainActivity.kt
package mchehab.com.kotlinapp

import kotlinx.android.synthetic.main.activity_main.*
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import io.realm.Realm

class MainActivity : AppCompatActivity() {

    private var person = Person()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if(isPersonSaved()){
            button.text = "Update"
            person = getPersonFromRealm()!!
        }

        editTextFirstName.setText(person.firstName)
        editTextLastName.setText(person.lastName)
        editTextAge.setText("${person.age} ")

        button.setOnClickListener {
            val realm = Realm.getDefaultInstance()
            realm.beginTransaction()

            person.firstName = editTextFirstName.text.toString()
            person.lastName = editTextLastName.text.toString()
            person.age = editTextAge.text.toString().trim().toInt()

            if(button.text.toString().equals("save", true)){
                realm.insert(person)
            }
            realm.commitTransaction()
        }

        buttonRestart.setOnClickListener {
            val intent = intent
            finish()
            startActivity(intent)
        }
    }

    private fun isPersonSaved(): Boolean {
        Realm.init(this)
        return Realm.getDefaultInstance().where(Person::class.java).findFirst() != null
    }

    private fun getPersonFromRealm(): Person? {
        Realm.init(this)
        val realm = Realm.getDefaultInstance()
        return realm.where(Person::class.java).findFirst()
    }
}